src/source-map.ts

```diff
@@ -37,6 +81,65 @@ export class SourceMapConsumer {
     return originalPositionFor(this._map, needle);
   }
 
+  generatedPositionFor(
+    originalPosition: Parameters<typeof generatedPositionFor>[1],
+  ): ReturnType<typeof generatedPositionFor> {
+    return generatedPositionFor(this._map, originalPosition);
+  }
+
+  allGeneratedPositionsFor(
+    originalPosition: Parameters<typeof generatedPositionFor>[1],
+  ): ReturnType<typeof generatedPositionFor>[] {
+    // This doesn't map exactly to the same feature
+    return [generatedPositionFor(this._map, originalPosition)];
+  }
+
+  hasContentsOfAllSources(): boolean {
+    if (!this.sourcesContent) {
+      return false;
+    }
+    return (
+      this.sourcesContent.length >= this.sources.length &&
+      !this.sourcesContent.some(function (sc) {
+        return sc == null;
+      })
+    );
+  }
+
+  sourceContentFor(aSource: string, nullOnMissing?: boolean): string | null {
+    if (!this.sourcesContent) {
+      return null;
+    }
+
+    const sourceContent = sourceContentFor(this._map, aSource);
+    if (sourceContent != null) {
+      return sourceContent;
+    }
+
+    const resolvedSource = resolve(aSource, this.sourceRoot);
+    if (this.sources.indexOf(resolvedSource) > -1) {
+      return this.sourcesContent[this.sources.indexOf(resolvedSource)];
+    }
+
+    if (nullOnMissing) {
+      return null;
+    } else {
+      throw new Error('"' + resolvedSource + '" is not in the SourceMap.');
+    }
+  }
+
+  eachMapping(
+    aCallback: Parameters<typeof eachMapping>[1],
+    aContext?: any /*, aOrder?: number*/,
```

Comment:
I don't know how important that `order` argument is, should it be implemented ? is it even possible in `trace-mapping` ?

Reply:
Documented that this isn't supported for now